rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/TurtleParser.java

```diff
@@ -680,19 +698,31 @@ protected Literal parseQuotedLiteral() throws IOException, RDFParseException, RD
 			// Read datatype
 			Value datatype = parseValue();
 			if (datatype instanceof IRI) {
-				return createLiteral(label, null, (IRI) datatype, getLineNumber(), -1);
-			} else {
+				return createLiteral(label, null, (IRI)datatype, getLineNumber(), -1);
+			}
+			else if (datatype != null) {
 				reportFatalError("Illegal datatype value: " + datatype);
 				return null;
 			}
-		} else {
+			else {
+				if (getParserConfig().get(BasicParserSettings.VERIFY_URI_SYNTAX)
```

Comment:
This logic should be available using AbstractRDFParser.reportError(String, RioSetting), which then checks whether it is a fatal error to throw an exception or return silently/log the error. This enables the ParseErrorListener to see the non-fatal error, which isn't the case in this code.

https://github.com/eclipse/rdf4j/blob/master/rio/api/src/main/java/org/eclipse/rdf4j/rio/helpers/AbstractRDFParser.java#L635

https://github.com/eclipse/rdf4j/blob/master/rio/api/src/main/java/org/eclipse/rdf4j/rio/helpers/RDFParserHelper.java#L359

Similar to this code:

https://github.com/eclipse/rdf4j/blob/master/rio/api/src/main/java/org/eclipse/rdf4j/rio/helpers/AbstractRDFParser.java#L387